egs/voxforge/asr1/run_rnnt.sh

```diff
@@ -0,0 +1,289 @@
+#!/bin/bash
+
+. ./path.sh || exit 1;
+. ./cmd.sh || exit 1;
+
+# general configuration
+backend=pytorch
+stage=-1       # start from -1 if you need to start from data download
+stop_stage=100
+ngpu=1         # number of gpus ("0" uses cpu, otherwise use gpu)
+debugmode=1
+dumpdir=dump   # directory to dump full features
+N=0            # number of minibatches to be used (mainly for debugging). "0" uses all minibatches.
+verbose=0      # verbose option
+resume=        # Resume the training from snapshot
+lm_resume=     # Resume the LM training from snapshot
+
+# data
+voxforge=downloads # original data directory to be stored
+lang=it # de, en, es, fr, it, nl, pt, ru
+
+# datasets
+train_set=tr_${lang}
+train_dev=dt_${lang}
+recog_set="dt_${lang} et_${lang}"
+
+# feature configuration
+do_delta=false
+
+# transducer configuration
+rnnt_mode='rnnt' # define transducer mode. Can be either 'rnnt' or 'rnnt-att'
+use_transfer=false # use transfer learning
+type_transfer='both' # define module to transfer. Can be either 'enc', 'dec' or 'both'
+
+## main configs
+train_config=conf/tuning/transducer/train_transducer.yaml
+decode_config=conf/tuning/transducer/decode_transducer.yaml
+## transfer learning configs
+ctc_config=conf/tuning/transducer/train_ctc.yaml
+att_config=conf/tuning/transducer/train_att.yaml
+lm_config=conf/tuning/transducer/lm_transducer.yaml
+
+if [ ${use_transfer} == true ]; then
+    exp_config=conf/tuning/transducer/train_transducer_${type_transfer}_init.yaml
+
+    sed '$a\' ${train_config} > ${exp_config}
+
+    case "${type_transfer}" in
+        enc|both)
+            echo "enc-init: 'exp/tr_it_pytorch_train_ctc/results/model.loss.best'" >> ${exp_config}
```

Comment:
Is it possible to change this `tr_it_pytorch_train_ctc` flexible (e.g., `tr_it_${backend}__$(basename ${ctc_config%.*}`)?
Then, we can change `ctc_config` to the other file.

Same comment for `train_rnnlm_pytorch_lm_transducer` and `tr_it_pytorch_train_att`.

Reply:
I suppose the other files you're talking about are the default rnn and lm configs, right? If so, I made different configs to match parameters between model (and also avoid user problems...).

I think I'll isolate this part. I'll add an utility script taking as argument main config, finetuning options and optional configs (CTC, Att and LM). It will create or modify the different needed configs with matched parameters, and return all configs paths. It will be easier to maintain like this and only the network training part would be different from the usual running script. What do you think @sw005320, @kan-bayashi?

Reply:
This sounds good. I think the pre-training -> fine-tuning becomes more important and such utility function helps to clean the script.

Reply:
I agree with your idea.
BTW, why don't you `utils/change_yaml.py` to change yaml file?
https://github.com/espnet/espnet#changing-the-configuration

Reply:
Because I was not aware of it! Thanks!
However it seems a bit overkill to use this script, no? Append/Delete entries to/from a file is rather straightforward with bash builtin and I don't think there is a gain in speed/memory.